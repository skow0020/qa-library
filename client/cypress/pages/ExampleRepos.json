{
    "qaRepos": "#row-0 [id*=repo-card]",
    "firstQARepo": "#row-0 #repo-card-0",
    "firstQARepoTitle": "#row-0 #repo-card-0 .card-title",
    "skow0020Repos": "#row-0 [id*=repo-card]",
    "firstskow0020Repo": "#row-0 #repo-card-0]",
    "language": "#language"
}