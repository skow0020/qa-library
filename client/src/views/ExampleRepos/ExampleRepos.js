/* eslint-disable react-hooks/exhaustive-deps */

import React, { useEffect, useState } from 'react';

import CardComponent from 'components/common/CardComponent';
import Chip from '@material-ui/core/Chip';
import Colors from 'utils/Colors';
import Grid from '@material-ui/core/Grid';
import LanguagesSelection from 'components/common/LanguagesSelection';
import LoadError from 'components/common/LoadError';
import Loading from 'components/common/Loading';
import PageTitle from 'components/common/PageTitle';
import TextField from 'components/common/TextField';
import Typography from '@material-ui/core/Typography';
import { getLanguageTheme } from 'utils/util';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  githubUser: {
    marginLeft: 'auto'
  }
}));

export default function ExampleRepos() {
  const classes = useStyles();
  const [githubAccount, setGithubAccount] = useState('skow0020');
  const [githubRepos, setGithubRepos] = useState([]);
  const [error, setError] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  const [filter, setFilter] = useState('');

  useEffect(() => {
    getRepos();
  }, []);

  useEffect(() => {
    getRepos();
  }, [filter]);

  const getRepos = () => {
    setIsLoading(true);
    setError(null);
    fetch(`https://api.github.com/users/${githubAccount}/repos`)
      .then(res => res.json())
      .then(
        (results) => {
          setIsLoading(false);
          if (results.message === 'Not Found') {
            setError('Unable to find repos for the account provided');
            return;
          }
          if (filter) {
            if (filter === 'CSharp') results = results.filter(repo => { return repo.language === 'C#'; });
            else results = results.filter(repo => { return repo.language === filter; });
          }

          setGithubRepos(results);
        },
        (error) => {
          setIsLoading(false);
          setError(error);
        }
      );
  };

  const handleGithubAccountSubmit = (e) => {
    e.preventDefault();
    setGithubAccount(githubAccount);
    getRepos();
  };

  return (
    <Grid container>
      <Grid container alignItems="center" >
        <PageTitle title="Example Repos" />
        <Typography id="github-account-text" className={classes.githubUser} variant="h5" color="textSecondary" align="right">
          {githubAccount} Repos
        </Typography>
      </Grid>
      <Grid container>
        <form id='filter-form' onSubmit={handleGithubAccountSubmit}>
          <TextField id="github-account" label="Github Account" value={githubAccount} onChange={(e) => setGithubAccount(e.target.value)} required />
          <LanguagesSelection id="language" value={filter} onChange={(e) => setFilter(e.target.value)} />
        </form>
      </Grid>
      {isLoading && <Loading />} 
      {error && <LoadError error={error} />}
      <Grid container spacing={4}>
        {githubRepos.map((repo, idx) => (
          <Grid item md={6} lg={4} sm={12} key={idx}>
            <CardComponent
              idx={`repo-card-${idx}`}
              url={repo.html_url}
              urlTarget="_blank"
              title={repo.name}
              avatar={<Chip
                size="small"
                label={repo.language}
                style={{ backgroundColor: repo.language ? getLanguageTheme(repo.language) : Colors.blue }}
              />}
              body={repo.description}
            />
          </Grid>
        ))}
      </Grid>
    </Grid>
  );
}