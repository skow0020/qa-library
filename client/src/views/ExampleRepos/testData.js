export const repos = [
  {
    'id': 198064850,
    'node_id': 'MDEwOlJlcG9zaXRvcnkxOTgwNjQ4NTA=',
    'name': 'acme-outreach',
    'full_name': 'skow0020/acme-outreach',
    'private': false,
    'owner': {
      'login': 'skow0020',
      'id': 12362090,
      'node_id': 'MDQ6VXNlcjEyMzYyMDkw',
      'avatar_url': 'https://avatars2.githubusercontent.com/u/12362090?v=4',
      'gravatar_id': '',
      'url': 'https://api.github.com/users/skow0020',
      'html_url': 'https://github.com/skow0020',
      'followers_url': 'https://api.github.com/users/skow0020/followers',
      'following_url': 'https://api.github.com/users/skow0020/following{/other_user}',
      'gists_url': 'https://api.github.com/users/skow0020/gists{/gist_id}',
      'starred_url': 'https://api.github.com/users/skow0020/starred{/owner}{/repo}',
      'subscriptions_url': 'https://api.github.com/users/skow0020/subscriptions',
      'organizations_url': 'https://api.github.com/users/skow0020/orgs',
      'repos_url': 'https://api.github.com/users/skow0020/repos',
      'events_url': 'https://api.github.com/users/skow0020/events{/privacy}',
      'received_events_url': 'https://api.github.com/users/skow0020/received_events',
      'type': 'User',
      'site_admin': false
    },
    'html_url': 'https://github.com/skow0020/acme-outreach',
    'description': null,
    'fork': false,
    'url': 'https://api.github.com/repos/skow0020/acme-outreach',
    'forks_url': 'https://api.github.com/repos/skow0020/acme-outreach/forks',
    'keys_url': 'https://api.github.com/repos/skow0020/acme-outreach/keys{/key_id}',
    'collaborators_url': 'https://api.github.com/repos/skow0020/acme-outreach/collaborators{/collaborator}',
    'teams_url': 'https://api.github.com/repos/skow0020/acme-outreach/teams',
    'hooks_url': 'https://api.github.com/repos/skow0020/acme-outreach/hooks',
    'issue_events_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues/events{/number}',
    'events_url': 'https://api.github.com/repos/skow0020/acme-outreach/events',
    'assignees_url': 'https://api.github.com/repos/skow0020/acme-outreach/assignees{/user}',
    'branches_url': 'https://api.github.com/repos/skow0020/acme-outreach/branches{/branch}',
    'tags_url': 'https://api.github.com/repos/skow0020/acme-outreach/tags',
    'blobs_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/blobs{/sha}',
    'git_tags_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/tags{/sha}',
    'git_refs_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/refs{/sha}',
    'trees_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/trees{/sha}',
    'statuses_url': 'https://api.github.com/repos/skow0020/acme-outreach/statuses/{sha}',
    'languages_url': 'https://api.github.com/repos/skow0020/acme-outreach/languages',
    'stargazers_url': 'https://api.github.com/repos/skow0020/acme-outreach/stargazers',
    'contributors_url': 'https://api.github.com/repos/skow0020/acme-outreach/contributors',
    'subscribers_url': 'https://api.github.com/repos/skow0020/acme-outreach/subscribers',
    'subscription_url': 'https://api.github.com/repos/skow0020/acme-outreach/subscription',
    'commits_url': 'https://api.github.com/repos/skow0020/acme-outreach/commits{/sha}',
    'git_commits_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/commits{/sha}',
    'comments_url': 'https://api.github.com/repos/skow0020/acme-outreach/comments{/number}',
    'issue_comment_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues/comments{/number}',
    'contents_url': 'https://api.github.com/repos/skow0020/acme-outreach/contents/{+path}',
    'compare_url': 'https://api.github.com/repos/skow0020/acme-outreach/compare/{base}...{head}',
    'merges_url': 'https://api.github.com/repos/skow0020/acme-outreach/merges',
    'archive_url': 'https://api.github.com/repos/skow0020/acme-outreach/{archive_format}{/ref}',
    'downloads_url': 'https://api.github.com/repos/skow0020/acme-outreach/downloads',
    'issues_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues{/number}',
    'pulls_url': 'https://api.github.com/repos/skow0020/acme-outreach/pulls{/number}',
    'milestones_url': 'https://api.github.com/repos/skow0020/acme-outreach/milestones{/number}',
    'notifications_url': 'https://api.github.com/repos/skow0020/acme-outreach/notifications{?since,all,participating}',
    'labels_url': 'https://api.github.com/repos/skow0020/acme-outreach/labels{/name}',
    'releases_url': 'https://api.github.com/repos/skow0020/acme-outreach/releases{/id}',
    'deployments_url': 'https://api.github.com/repos/skow0020/acme-outreach/deployments',
    'created_at': '2019-07-21T14:13:49Z',
    'updated_at': '2019-07-21T14:37:03Z',
    'pushed_at': '2019-07-21T14:37:01Z',
    'git_url': 'git://github.com/skow0020/acme-outreach.git',
    'ssh_url': 'git@github.com:skow0020/acme-outreach.git',
    'clone_url': 'https://github.com/skow0020/acme-outreach.git',
    'svn_url': 'https://github.com/skow0020/acme-outreach',
    'homepage': null,
    'size': 164,
    'stargazers_count': 0,
    'watchers_count': 0,
    'language': 'JavaScript',
    'has_issues': true,
    'has_projects': true,
    'has_downloads': true,
    'has_wiki': true,
    'has_pages': false,
    'forks_count': 0,
    'mirror_url': null,
    'archived': false,
    'disabled': false,
    'open_issues_count': 0,
    'license': null,
    'forks': 0,
    'open_issues': 0,
    'watchers': 0,
    'default_branch': 'master'
  },
  {
    'id': 85446048,
    'node_id': 'MDEwOlJlcG9zaXRvcnk4NTQ0NjA0OA==',
    'name': 'Appium-Tutorial',
    'full_name': 'skow0020/Appium-Tutorial',
    'private': false,
    'owner': {
      'login': 'skow0020',
      'id': 12362090,
      'node_id': 'MDQ6VXNlcjEyMzYyMDkw',
      'avatar_url': 'https://avatars2.githubusercontent.com/u/12362090?v=4',
      'gravatar_id': '',
      'url': 'https://api.github.com/users/skow0020',
      'html_url': 'https://github.com/skow0020',
      'followers_url': 'https://api.github.com/users/skow0020/followers',
      'following_url': 'https://api.github.com/users/skow0020/following{/other_user}',
      'gists_url': 'https://api.github.com/users/skow0020/gists{/gist_id}',
      'starred_url': 'https://api.github.com/users/skow0020/starred{/owner}{/repo}',
      'subscriptions_url': 'https://api.github.com/users/skow0020/subscriptions',
      'organizations_url': 'https://api.github.com/users/skow0020/orgs',
      'repos_url': 'https://api.github.com/users/skow0020/repos',
      'events_url': 'https://api.github.com/users/skow0020/events{/privacy}',
      'received_events_url': 'https://api.github.com/users/skow0020/received_events',
      'type': 'User',
      'site_admin': false
    },
    'html_url': 'https://github.com/skow0020/Appium-Tutorial',
    'description': 'Personal project for learning Appium - In Progress',
    'fork': false,
    'url': 'https://api.github.com/repos/skow0020/Appium-Tutorial',
    'forks_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/forks',
    'keys_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/keys{/key_id}',
    'collaborators_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/collaborators{/collaborator}',
    'teams_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/teams',
    'hooks_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/hooks',
    'issue_events_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues/events{/number}',
    'events_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/events',
    'assignees_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/assignees{/user}',
    'branches_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/branches{/branch}',
    'tags_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/tags',
    'blobs_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/blobs{/sha}',
    'git_tags_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/tags{/sha}',
    'git_refs_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/refs{/sha}',
    'trees_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/trees{/sha}',
    'statuses_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/statuses/{sha}',
    'languages_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/languages',
    'stargazers_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/stargazers',
    'contributors_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/contributors',
    'subscribers_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/subscribers',
    'subscription_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/subscription',
    'commits_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/commits{/sha}',
    'git_commits_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/commits{/sha}',
    'comments_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/comments{/number}',
    'issue_comment_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues/comments{/number}',
    'contents_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/contents/{+path}',
    'compare_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/compare/{base}...{head}',
    'merges_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/merges',
    'archive_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/{archive_format}{/ref}',
    'downloads_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/downloads',
    'issues_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues{/number}',
    'pulls_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/pulls{/number}',
    'milestones_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/milestones{/number}',
    'notifications_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/notifications{?since,all,participating}',
    'labels_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/labels{/name}',
    'releases_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/releases{/id}',
    'deployments_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/deployments',
    'created_at': '2017-03-19T02:36:41Z',
    'updated_at': '2017-03-20T14:03:03Z',
    'pushed_at': '2017-03-31T01:34:23Z',
    'git_url': 'git://github.com/skow0020/Appium-Tutorial.git',
    'ssh_url': 'git@github.com:skow0020/Appium-Tutorial.git',
    'clone_url': 'https://github.com/skow0020/Appium-Tutorial.git',
    'svn_url': 'https://github.com/skow0020/Appium-Tutorial',
    'homepage': '',
    'size': 126,
    'stargazers_count': 0,
    'watchers_count': 0,
    'language': 'Java',
    'has_issues': true,
    'has_projects': true,
    'has_downloads': true,
    'has_wiki': true,
    'has_pages': false,
    'forks_count': 0,
    'mirror_url': null,
    'archived': false,
    'disabled': false,
    'open_issues_count': 0,
    'license': null,
    'forks': 0,
    'open_issues': 0,
    'watchers': 0,
    'default_branch': 'master'
  }
];
