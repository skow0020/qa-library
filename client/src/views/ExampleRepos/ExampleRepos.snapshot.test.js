import ExampleRepos from './ExampleRepos';
import React from 'react';
import renderer from 'react-test-renderer';

describe('ExampleRepos Snapshot Tests', () => {
  test('ExampleRepos snapshot', () => {
    const component = renderer.create(
      <ExampleRepos></ExampleRepos>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});