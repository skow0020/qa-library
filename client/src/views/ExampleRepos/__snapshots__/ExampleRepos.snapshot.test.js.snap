// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleRepos Snapshot Tests ExampleRepos snapshot 1`] = `
<div
  className="main-content-container px-4 container-fluid"
>
  <div
    className="form-inline py-2 no-gutters row"
  >
    <div
      className="text-sm-left text-center text-md-left mb-sm-0 col-12 col-sm-8"
    >
      <span
        className="text-uppercase page-subtitle"
      />
      <h3
        className="page-title"
      >
        Example Repos
      </h3>
    </div>
    <form
      className=""
      id="filter-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="text-muted font-weight-bold px-2"
          htmlFor="github-account"
        >
          Github Account
        </label>
        <input
          className="form-control"
          id="github-account"
          onChange={[Function]}
          required={true}
          type="text"
          value="skow0020"
        />
        <div
          className="col"
        >
          <label
            className="text-muted font-weight-bold px-2"
            htmlFor="language"
          >
            Languages
          </label>
          <select
            className="form-control custom-select"
            id="language"
            onChange={[Function]}
            type="text"
          >
            <option />
            <option>
              Java
            </option>
            <option>
              CSharp
            </option>
            <option>
              Python
            </option>
            <option>
              Ruby
            </option>
            <option>
              JavaScript
            </option>
            <option>
              Cpp
            </option>
            <option>
              Swift
            </option>
          </select>
        </div>
      </div>
    </form>
  </div>
  <span
    className="text-uppercase page-subtitle"
  >
    skow0020
     Repos
  </span>
  <div
    className="row"
    id="row-0"
  />
</div>
`;
